android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -392,21 +391,41 @@ constructor(
     defaultRepository.addOrUpdate(resource = questionnaireResponse)
   }
 
-  suspend fun performExtraction(
+  fun performExtraction(
     context: Context,
     questionnaire: Questionnaire,
     questionnaireResponse: QuestionnaireResponse
   ): Bundle {
-
-    return ResourceMapper.extract(
-      questionnaire = questionnaire,
-      questionnaireResponse = questionnaireResponse,
-      StructureMapExtractionContext(
-        context = context,
-        transformSupportServices = transformSupportServices,
-        structureMapProvider = retrieveStructureMapProvider()
-      )
-    )
+    var bundle = Bundle()
+    viewModelScope.launch(dispatcherProvider.main()) {
+      kotlin
+        .runCatching {
+          withContext(dispatcherProvider.io()) {
+            ResourceMapper.extract(
+              questionnaire = questionnaire,
+              questionnaireResponse = questionnaireResponse,
+              StructureMapExtractionContext(
+                context = context,
+                transformSupportServices = transformSupportServices,
+                structureMapProvider = retrieveStructureMapProvider()
+              )
+            )
+          }
+        }
+        .onSuccess {
+          bundle = it
+          context.showToast(context.getString(R.string.structure_success))
+        }
+        .onFailure { exception ->
+          Timber.e(exception)
+          if (exception.message!!.contains("StructureMap")) {
+            context.showToast(context.getString(R.string.structure_map_missing_message))
+          } else {
+            context.showToast(context.getString(R.string.structure_error_message))
```

Comment:
can we mention the specific structuremap that had a problem in the error?

Reply:
If a `StructureMap` exists but extraction Fails, other exceptions are thrown. This is the reason the second check. To handle that.